### PR TITLE
Add default value for assessment

### PR DESCRIPTION
**Pull request by jonolehagemo**

### CODE REVIEW

File: src/datadoc/backend/datadoc_metadata.py [hunk 1]
```diff
@@ -104,6 +106,7 @@ def extract_metadata_from_files(self) -> None:
         elif self.dataset is not None:
             self.extract_metadata_from_dataset(self.dataset)
             self.meta.dataset.id = uuid.uuid4()
+            self.meta.dataset.assessment = Assessment.PROTECTED
```
Comment (mmwinther) [inline src/datadoc/backend/datadoc_metadata.py:109]: Does this not override the logic on L190-199?

File: src/datadoc/backend/datadoc_metadata.py [hunk 2]
```diff
@@ -184,6 +187,16 @@ def extract_metadata_from_dataset(
             short_name=self.short_name,
             dataset_state=dapla_dataset_path_info.dataset_state,
             dataset_status=DatasetStatus.DRAFT,
+            assessment=Assessment.PROTECTED
```
Comment (mmwinther) [inline src/datadoc/backend/datadoc_metadata.py:190]: I think this section could be more readable if we implemented it as a mapping, similar to this example from Ruff: https://docs.astral.sh/ruff/rules/too-many-return-statements/#example

File: tests/backend/test_datadoc_metadata.py [hunk 3]
```diff
@@ -262,6 +263,46 @@ def test_dataset_status_default_value(
     assert expected_type == datadoc_metadata.meta.dataset.dataset_status
 
 
+@pytest.mark.parametrize(
```
Comment (mmwinther) [inline tests/backend/test_datadoc_metadata.py:266]: We should add some more cases to this test, to cover all the possible values of `Assessment`.

**Review by mmwinther — CHANGES_REQUESTED** [review]
Good start, I think it needs a little bit more work, see the review comments 👍

File: src/datadoc/backend/datadoc_metadata.py (hunk 2 above)
Comment (jonolehagemo) [reply src/datadoc/backend/datadoc_metadata.py:186]: Refactored unreadable code into a static method to increase readability

File: src/datadoc/backend/datadoc_metadata.py (hunk 1 above)
Comment (jonolehagemo) [reply src/datadoc/backend/datadoc_metadata.py]: This line is removed

**Comment by jonolehagemo** [comment]
@mmwinther 
I haven't figured out how to refactor the global state to use dependecy injection instead. Any ideas?

**Review by mmwinther — DISMISSED** [review]
Looks good, nice work!

**Review by mmwinther — APPROVED** [review]

**Pull request merged**